src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveModuleBuilderTests.cs

```diff
@@ -0,0 +1,260 @@
+
+            foreach (int size in new int[] { -1, 0, 0x003f0000, 0x003f0000 + 1 })
+            {
+                AssertExtensions.Throws<ArgumentException>("size", () => module.DefineUninitializedData("TestField", size, FieldAttributes.Private));
+            }
+
+            module.CreateGlobalFunctions();
+            Assert.Throws<InvalidOperationException>(() => module.DefineUninitializedData("TestField", 1, FieldAttributes.HasDefault));
+        }
+
+        [Theory]
+        [InlineData(FieldAttributes.Static | FieldAttributes.Public)]
+        [InlineData(FieldAttributes.Static | FieldAttributes.Private)]
+        [InlineData(FieldAttributes.Private)]
+        public void DefineInitializedDataTest(FieldAttributes attributes)
+        {
+            AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderAndSaveMethod(new AssemblyName("MyAssembly"), out MethodInfo _);
+            ModuleBuilder module = ab.DefineDynamicModule("MyModule");
+            FieldBuilder field = module.DefineInitializedData("MyField", [01, 00, 01], attributes);
+
+            Assert.True(field.IsStatic);
+            Assert.Equal((attributes & FieldAttributes.Public) != 0, field.IsPublic);
+            Assert.Equal((attributes & FieldAttributes.Private) != 0, field.IsPrivate);
+            Assert.Equal("MyField", field.Name);
+        }
+
+        [Fact]
+        public void DefineInitializedData_Validations()
+        {
+            AssemblyBuilder ab = AssemblySaveTools.PopulateAssemblyBuilderAndSaveMethod(new AssemblyName("MyAssembly"), out MethodInfo _);
+            ModuleBuilder module = ab.DefineDynamicModule("MyModule");
+
+            AssertExtensions.Throws<ArgumentNullException>("name", () => module.DefineInitializedData(null, [1, 0, 1], FieldAttributes.Public));
+            AssertExtensions.Throws<ArgumentException>("name", () => module.DefineInitializedData("", [1, 0, 1], FieldAttributes.Private));
+            AssertExtensions.Throws<ArgumentNullException>("data", () => module.DefineInitializedData("MyField", null, FieldAttributes.Public));
+            AssertExtensions.Throws<ArgumentException>("size", () => module.DefineInitializedData("MyField", [], FieldAttributes.Public));
+            AssertExtensions.Throws<ArgumentException>("size", () => module.DefineInitializedData("MyField", new byte[0x3f0000], FieldAttributes.Public));
+
+            FieldBuilder field = module.DefineInitializedData("MyField", [1, 0, 1], FieldAttributes.Public);
+            module.CreateGlobalFunctions();
+
+            Assert.Null(field.DeclaringType);
+            Assert.Throws<InvalidOperationException>(() => module.DefineInitializedData("MyField2", new byte[] { 1, 0, 1 }, FieldAttributes.Public));
+        }
+
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void DefineInitializedData_EnsureAlignmentIsMinimumNeededForUseOfCreateSpan()
+        {
+            RemoteExecutor.Invoke(() =>
```

Comment:
Is there a reason we need to run this in a separate process? The Loader should be resliliant against malformed assemblies and refuse to load them.

Reply:
The reason Steve mentioned in https://github.com/dotnet/runtime/pull/96517#discussion_r1442424016
`The assembly won't be unloaded, so over time such tests will bloat the process.`

Reply:
This is predicated on using `LoadFrom()`, which we should avoid in most cases. An unloadable alternative that I think would be much preferrable is to use a collectable ALC and that way we can avoid the remote process approach and it is more inline with how we should be testing assembly loading.